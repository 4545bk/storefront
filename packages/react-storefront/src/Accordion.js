/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { useState, useContext } from 'react'
import AppContext from './AppContext'
import AmpAccordion from './amp/AmpAccordion'

const AccordionContext = React.createContext()

export const withAccordionContext = Component => {
  return props => (
    <AccordionContext.Consumer>
      {ctx => <Component {...props} {...ctx} />}
    </AccordionContext.Consumer>
  )
}

/**
 * Accordion which only allows one child ExpandableSection to be open at a time
 *
 * ```js
 *  <Accordion>
 *    <ExpandableSection title="First">
 *      <div>The first section</div>
 *    </ExpandableSection>
 *    <ExpandableSection title="Second">
 *      <div>The second section</div>
 *    </ExpandableSection>
 *    <ExpandableSection title="Third">
 *      <div>The third section</div>
 *    </ExpandableSection>
 *  </Accordion>
 * ```
 */
export default function Accordion({ children, ...otherProps }) {
  const [expandedSectionId, setExpandedSectionId] = useState(null)
  const {
    app: { amp }
  } = useContext(AppContext)
  return (
    <AccordionContext.Provider
      value={{
        expandedSectionId,
        setExpandedSectionId
      }}
    >
      {amp ? <AmpAccordion {...otherProps}>{children}</AmpAccordion> : children}
    </AccordionContext.Provider>
  )
}
