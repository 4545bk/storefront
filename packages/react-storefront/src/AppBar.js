/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { throttle } from 'lodash'
import Toolbar from '@material-ui/core/Toolbar'
import SvgIcon from '@material-ui/core/SvgIcon'
import { observer, inject } from 'mobx-react'
import Hidden from '@material-ui/core/Hidden'
import withStyles from '@material-ui/core/styles/withStyles'
import { Vbox } from './Box'
import ToolbarButton from './ToolbarButton'
import MenuIcon from './assets/menu.svg'

export const styles = (theme) => ({
  root: {
    boxSizing: 'content-box',
    position: 'relative',
    zIndex: theme.zIndex.modal + 10
  },

  withAmp: {
    zIndex: theme.zIndex.amp.modal + 1
  },

  icon: {
    color: theme.palette.action.active
  },

  toolBar: {
    height: theme.headerHeight || '64px',
    padding: `0 7px`,
    maxWidth: theme.maxWidth,
    flex: 1
  },

  wrap: {
    borderBottom: `1px solid ${theme.palette.divider}`,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    backgroundColor: theme.palette.background.paper,
    zIndex: theme.zIndex.modal + 10,
    display: 'flex',
    justifyContent: 'center',
  },

  stuck: {
    transform: 'translateY(0)',
  },

  unstuck: {
    transform: 'translateY(-100%)',
  },

  animate: {
    transition: 'transform .15s ease-in',
  },

  hidden: {
    position: 'fixed',
    zIndex: theme.zIndex.modal + 10,
    boxShadow: theme.shadows[2],
    top: 0,
    left: 0,
    right: 0
  },

  menuOpen: {
    boxShadow: 'none'
  },

  link: {
    textDecoration: 'none'
  }
});

/**
 * A header that auto hides when the user scrolls down and auto shows when the user 
 * scrolls up. A hamburger button that controls the menu is automatically displayed on
 * the left side of the header.  Children are placed directly to the right of the menu
 * button.
 */
@inject(({ app }) => ({ menu: app.menu, amp: app.amp }))
@withStyles(styles, { name: 'RSFAppBar'})
@observer
export default class Header extends Component {

  state = {
    stuck: false,
    hidden: false,
    animate: false
  }

  static propTypes = {
    /**
     * CSS classes
     */
    classes: PropTypes.object,

    /**
     * Set to true to show the hamburger button only on mobile devices. Defaults to false.
     */
    responsive: PropTypes.bool,

    /**
     * A component for the menu icon.
     */
    MenuIcon: PropTypes.func
  }

  static defaultProps = {
    responsive: false,
    MenuIcon
  }

  render() {
    const { MenuIcon, classes, children, fixed, menu, responsive, width, amp, showLabels } = this.props

    const menuIcon = <SvgIcon className={classes.icon}><MenuIcon/></SvgIcon>

    return (
      <div className={classnames({ [classes.root]: true, [classes.withAmp]: amp })}> 
        <div className={classnames({
          [classes.wrap]: true,
          [classes.fixed]: fixed,
          [classes.hidden]: this.state.hidden || menu.open,
          [classes.stuck]: (this.state.hidden && this.state.stuck) || menu.open ,
          [classes.unstuck]: this.state.hidden && !this.state.stuck,
          [classes.animate]: this.state.animate,
          [classes.menuOpen]: menu.open
        })}>
          <Toolbar disableGutters classes={{ root: classes.toolBar }}>
            <Hidden mdUp={responsive} implementation="css">
              <a on="tap:moov_menu.toggle" className={classes.link}>
                <ToolbarButton 
                  aria-label="Menu" 
                  color="inherit" 
                  onClick={menu.toggle}
                  icon={menuIcon}
                />
              </a>
            </Hidden>
            { children }
          </Toolbar>
        </div>
      </div>
    )
  }

  componentDidMount() {
    window.addEventListener('scroll', this.onScroll, { passive: true })
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.onScroll, { passive: true })
  }

  componentDidUpdate(prevProps, prevState) {
    if (!prevState.hidden && this.state.hidden) {
      setTimeout(() => this.setState({ animate: true }), 100)
    }
  }

  onScroll = () => {
    const height = 64, { scrollY } = window, { lastScrollY } = this
    const { menu } = this.props

    if (this.state.hidden) {
      if (scrollY <= 0) {
        this.setState({ hidden: false, stuck: false, animate: false })
      }
    } else {
      if (scrollY > height && !menu.open) {
        this.setState({ hidden: true })
      }
    }

    const unstickBufferZone = 15

    if (this.throttledScrollY > scrollY + unstickBufferZone && !this.state.stuck) {
      this.unstickAt = scrollY + unstickBufferZone
      this.setState({ stuck: true })
    } else if (scrollY > this.unstickAt && this.state.stuck && !menu.open) {
      this.setState({ stuck: false })
      delete this.unstickAt
    }

    if (this.lastScrollY > scrollY && this.state.stuck) {
      this.unstickAt = scrollY + unstickBufferZone
    }

    this.lastScrollY = scrollY
    this.sampleScrollSpeed()
  }

  sampleScrollSpeed = throttle(() => this.throttledScrollY = window.scrollY, 100)

}
