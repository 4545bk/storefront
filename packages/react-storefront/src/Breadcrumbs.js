/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import ArrowRight from '@material-ui/icons/KeyboardArrowRight'
import Link from './Link'
import withStyles from '@material-ui/core/styles/withStyles'
import Typography from '@material-ui/core/Typography'
import { inject, observer } from 'mobx-react'
import Track from './Track'
import Container from './Container'

export const styles = theme => ({
  breadcrumbs: {
    margin: '0',
    backgroundColor: '#F4F2F1',
    padding: '12px 0',

    '& a': {
      color: theme.palette.text.primary,
      textDecoration: 'none'
    }
  },

  separator: {
    height: '12px',
    position: 'relative',
    top: '3px',
    width: '16px'
  },

  current: {
    fontWeight: 'bold',
    color: theme.palette.text.primary
  }
})

/**
 * Renders a list of breadcrumbs from AppModelBase.breadcrumbs.  When a breadcrumb is clicked, the breadcrumb_clicked
 * analytics event is automatically fired.
 */
@withStyles(styles, { name: 'RSFBreadcrumbs' })
@inject('app')
@observer
export default class Breadcrumbs extends Component {
  render() {
    const { app: { breadcrumbs }, classes } = this.props

    if (!breadcrumbs || !breadcrumbs.length) return null

    return (
      <Typography className={classes.breadcrumbs} variant="caption">
        <Container>
          { breadcrumbs.map(this.renderBreadcrumb) }
        </Container>
      </Typography>
    )
  }

  renderBreadcrumb = (item, i) => {
    const { classes } = this.props
    
    if (item.url) {
      return (
        <span key={i}>
          <Track event="breadcrumbClicked" breadcrumb={item}>
            <Link to={item.url}>{item.text}</Link>
          </Track>
          <ArrowRight className={classes.separator}/>
        </span>
      )
    } else {
      return (
        <span key={i} className={classes.current}>{item.text}</span>
      )
    }
  }

}
