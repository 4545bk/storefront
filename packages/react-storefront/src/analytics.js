/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */

let _targets = []

/**
 * Configures the PWA to broadcast events to all specified targets.  Once configured, You can all any method on
 * the the default import from this module it will attempt to call a method of the same name on each target.  
 * 
 * Example:
 * 
 *  // src/analytics.js
 * 
 *  import { configureAnalytics } from 'react-storefront/analytics'
 *  import GoogleTagManagerTarget from 'moov-pwa-analytics/GoogleTagManagerTarget'
 *  import MerkleSearchTarget from 'moov-pwa-analytics/MerkleSearchTarget'
 * 
 *  configureAnalytics(
 *    new GoogleTagManagerTarget({ apiKey: 'my_gtm_api_key' }),
 *    new MerkleSearchTarget({ apiKey: 'my_merkle_api_key' })
 *  )
 * 
 * // then, to broadcast an analytics event to all targets:
 * 
 * import analytics from 'react-storefront/analytics'
 * 
 * analytics.someEvent() // this will call the 'someEvent' method on all configured targets
 * 
 * @param {AnalyticsTarget[]} targets An array of targets to notify when analytics events occur
 * @return {AnalyticsProvider}
 */
export function configureAnalytics(...targets) {
  _targets = targets
}

export function getTargets() {
  return _targets
}

/**
 * This module's default export is a single entry point for broadcasting analytics events to all targets configured via `configureAnalytics`.
 * Each target represents a tag manager or other analytics service to which the application should send data.
 * 
 * Example:
 * 
 *  // src/analytics.js
 * 
 *  import { configureAnalytics } from 'react-storefront/analytics'
 *  import GoogleTagManagerTarget from 'moov-pwa-analytics/GoogleTagManagerTarget'
 *  import MerkleSearchTarget from 'moov-pwa-analytics/MerkleSearchTarget'
 * 
 *  configureAnalytics(
 *    new GoogleTagManagerTarget({ apiKey: 'my_gtm_api_key' }),
 *    new MerkleSearchTarget({ apiKey: 'my_merkle_api_key' })
 *  )
 * 
 */
export default new Proxy(
  { 
    setHistory: () => {} // polyfill for IE11 requires defined fields in Target argument
  }, 
  {
    get: function (o, method) {
      return function(...args) {
        for (let target of _targets) {
          const fn = target[method]

          if (fn) {
            fn.apply(target, args)
          } else {
            if (typeof method === 'string') {
              console.warn(`${target.constructor.name} does not support ${method}`)
            }
          }
        }
      }
    }
  }
)
