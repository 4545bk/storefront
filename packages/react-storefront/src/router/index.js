/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
export { default as Router } from './Router'
export { default as fromServer } from './fromServer'
export { default as fromClient } from './fromClient'
export { default as proxyUpstream } from './proxyUpstream'
export { default as cache } from './cache'
export { default as propagateCookies } from './propagateCookies'
export { default as withGlobalState } from './withGlobalState'
export { default as Response } from './Response'
