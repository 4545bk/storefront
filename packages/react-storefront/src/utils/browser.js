/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
let _isSafari = null

/**
 * Clears cached results.  Only needed for unit testing.
 * @private
 */
export function clearTestCache() {
  _isSafari = null
}

/**
 * Returns true only in Safari
 * @return {Boolean}
 */
export function isSafari() {
  if (_isSafari == null) {
    _isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent)
  }
  return _isSafari
}
