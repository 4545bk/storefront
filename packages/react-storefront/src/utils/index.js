// @create-index

export { default as browser } from './browser.js';
export { default as createGenerateClassName } from './createGenerateClassName.js';
export { default as dataProps } from './dataProps.js';
export { default as scriptInjector } from './scriptInjector.js';
export { default as url } from './url.js';

