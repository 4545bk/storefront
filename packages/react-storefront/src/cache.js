/**
 * @license
 * Copyright Â© 2017-2019 Moov Corporation.  All rights reserved.
 */
import { clearCache } from './router/serviceWorker'

/**
 * Clears all API and SSR responses from the client cache
 * @return {Promise} resolved once the cache has been cleared
 */
export function clearClientCache() {
  return clearCache()
}
