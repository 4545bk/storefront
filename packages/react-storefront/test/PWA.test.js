/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'mobx-react'
import AppModelBase from '../src/model/AppModelBase'
import PWA from '../src/PWA'
import simulant from 'simulant'
import { clearTestCache } from '../src/utils/browser'
import { Router, proxyUpstream } from '../src/router';

describe('PWA', () => {
  let history, app, userAgent

  beforeEach(() => {
    jest.spyOn(global.console, 'error').mockImplementation()
    jest.spyOn(window.navigator, 'userAgent', 'get').mockImplementation(() => userAgent)
    app = AppModelBase.create({ location: { hostname: 'localhost', pathname: '/', search: '' } })
    history = { push: jest.fn(), listen: jest.fn() }
  })

  it('should render amp-install-service worker when amp==true', () => {
    const wrapper = mount(
      <Provider history={history} app={AppModelBase.create({ amp: true, location: { hostname: 'localhost', pathname: '/', search: '' } })}>
        <PWA>
          <div>Foo</div>
        </PWA>
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should call history.push when a link to the same domain is clicked', () => {
    const wrapper = mount(
      <Provider history={history} app={app}>
        <PWA>
          <a href="/foo">Foo</a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    simulant.fire(document.body.querySelector('a'), 'click')

    expect(history.push).toHaveBeenCalledWith('/foo')
  })

  it('should not call history.push when the link has a target other than _self', () => {
    const wrapper = mount(
      <Provider history={history} app={app}>
        <PWA>
          <a href="/foo" target="_blank">Foo</a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    simulant.fire(document.body.querySelector('a'), 'click')

    expect(history.push).not.toHaveBeenCalled()
  })

  it('should call history.push when a descendant element of a link is clicked', () => {
    const wrapper = mount(
      <Provider history={history} app={app}>
        <PWA>
          <a href="/foo"><p id="target">Foo</p></a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    simulant.fire(document.body.querySelector('#target'), 'click')

    expect(history.push).toHaveBeenCalledWith('/foo')
  })

  it('should not call history.push when the link has mailto:', () => {
    const wrapper = mount(
      <Provider history={history} app={app}>
        <PWA>
          <a href="mailto:user@domain.com">Foo</a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    simulant.fire(document.body.querySelector('a'), 'click')

    expect(history.push).not.toHaveBeenCalled()
  })

  it('should not call history.push when the link has tel:', () => {
    const wrapper = mount(
      <Provider history={history} app={app}>
        <PWA>
          <a href="tel:1111111111">Foo</a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    simulant.fire(document.body.querySelector('a'), 'click')

    expect(history.push).not.toHaveBeenCalled()
  })

  it('should not history.push when the link has target=_self', () => {
    const wrapper = mount(
      <Provider history={history} app={app}>
        <PWA>
          <a href="/foo" target="_self">Foo</a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    simulant.fire(document.body.querySelector('a'), 'click')

    expect(history.push).toHaveBeenCalledWith('/foo')
  })

  it('should not call history.push when a link to another domain is clicked', () => {
    const wrapper = mount(
      <Provider history={history} app={app}>
        <PWA>
          <a id="link1" href="http://www.google.com">Google 1</a>
          <a id="link2" href="https://www.google.com">Google 2</a>
          <a id="link3" href="//www.google.com">Google 3</a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    document.body.querySelectorAll('a').forEach(a => simulant.fire(a, 'click'))

    expect(history.push.mock.calls.length).toEqual(0)
  })

  it('should not call history.push when a link points to a route with a proxyUpstream handler', () => {
    const router = new Router()
      .get('/proxy', proxyUpstream('./proxyHandler'))
      
    const wrapper = mount(
      <Provider history={history} app={app} router={router}>
        <PWA>
          <a href="/proxy">proxyUpstream</a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    document.body.querySelectorAll('a').forEach(a => simulant.fire(a, 'click'))

    expect(history.push.mock.calls.length).toEqual(0)
  })

  it('should render children', () => {
    expect(mount(
      <Provider history={history} app={app}>
        <PWA>
          <div>foo</div>
        </PWA>
      </Provider>
    )).toMatchSnapshot()
  })

  it('should reload the page when data-reload="on"', () => {
    const wrapper = mount(
      <Provider history={history} app={app}>
        <PWA>
          <a href="/foo" data-reload="on">Foo</a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    simulant.fire(document.body.querySelector('a'), 'click')

    expect(history.push.mock.calls.length).toEqual(0)
  })

  it('should reload the page when data-reload="true"', () => {
    const wrapper = mount(
      <Provider history={history} app={app}>
        <PWA>
          <a href="/foo" data-reload="true">Foo</a>
        </PWA>
      </Provider>
    , { attachTo: document.body })

    simulant.fire(document.body.querySelector('a'), 'click')

    expect(history.push.mock.calls.length).toEqual(0)
  })

  it('should add the moov-safari class to the body when in safari', () => {
    clearTestCache()
    document.body.classList.remove('moov-safari')
    userAgent = "Mozilla/5.0 (iPhone; CPU iPhone OS 11_0 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/11.0 Mobile/15A372 Safari/604.1"

    mount(
      <Provider history={history} app={app}>
        <PWA/>
      </Provider>
    , { attachTo: document.body })

    expect(document.body.classList.contains('moov-safari')).toBe(true)
  })

  it('should not add the moov-safari class to the body when not in safari', () => {
    clearTestCache()
    document.body.classList.remove('moov-safari')
    userAgent = "Mozilla/5.0 (Linux; Android 5.0; SM-G900P Build/LRX21T) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Mobile Safari/537.36"

    mount(
      <Provider history={history} app={app}>
        <PWA/>
      </Provider>
    , { attachTo: document.body })

    expect(document.body.classList.contains('moov-safari')).toBe(false)
  })

  it('should catch errors during rendering and display the error view', () => {
    const RenderError = () => {
      throw new Error("Error during rendering")
    }

    mount(
      <Provider history={history} app={app}>
        <PWA>
          <RenderError/>
        </PWA>
      </Provider>
    )

    expect(app.error).toBe("Error during rendering")
    expect(app.stack).not.toBeNull()
    expect(app.page).toBe('Error')
  })
})