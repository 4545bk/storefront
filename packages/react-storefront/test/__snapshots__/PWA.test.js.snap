// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PWA should render amp-install-service worker when amp==true 1`] = `
<Provider history={{...}} app={{...}}>
  <WithStyles(inject-PWA)>
    <inject-PWA classes={{...}}>
      <PWA classes={{...}} menu={{...}} app={{...}} history={{...}} amp={true}>
        <Provider nextId={[Function]}>
          <WithStyles(CssBaseline)>
            <CssBaseline classes={{...}} />
          </WithStyles(CssBaseline)>
          <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
            <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} title={{...}} titleAttributes={{...}} htmlAttributes={{...}} meta={{...}}>
              <NullComponent defer={true} encodeSpecialCharacters={true} title={{...}} titleAttributes={{...}} htmlAttributes={{...}} meta={{...}} />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
          <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
            <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
              <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
          <amp-install-serviceworker src="https://localhost/service-worker.js" data-iframe-src="https://localhost/pwa/install-service-worker.html" layout="nodisplay" />
          <div>
            Foo
          </div>
        </Provider>
      </PWA>
    </inject-PWA>
  </WithStyles(inject-PWA)>
</Provider>
`;

exports[`PWA should render children 1`] = `
<Provider history={{...}} app={{...}}>
  <WithStyles(inject-PWA)>
    <inject-PWA classes={{...}}>
      <PWA classes={{...}} menu={{...}} app={{...}} history={{...}} amp={false}>
        <Provider nextId={[Function]}>
          <WithStyles(CssBaseline)>
            <CssBaseline classes={{...}} />
          </WithStyles(CssBaseline)>
          <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
            <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} title={{...}} titleAttributes={{...}} htmlAttributes={{...}} meta={{...}}>
              <NullComponent defer={true} encodeSpecialCharacters={true} title={{...}} titleAttributes={{...}} htmlAttributes={{...}} meta={{...}} />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
          <div>
            foo
          </div>
        </Provider>
      </PWA>
    </inject-PWA>
  </WithStyles(inject-PWA)>
</Provider>
`;
